app/models/step_by_step_page.rb

```diff
@@ -41,6 +41,21 @@ def self.validate_redirect(redirect_url)
     redirect_url =~ regex
   end
 
+  # This function first generates a hash from the object id,
+  # then unpacks it to integers,
+  # and then finally interpolates them into a string.
+  # The purpose is to generate a deterministic hash
+  # It mostly rips off SecureRandom.uuid
+  # see http://ruby-doc.org/stdlib-1.9.3/libdoc/securerandom/rdoc/SecureRandom.html#uuid-method
+  def auth_bypass_id
+    @_auth_bypass_id ||= begin
+      ary = Digest::SHA256.hexdigest(id.to_s).unpack('NnnnnN')
```

Comment:
I wonder about `id.to_s` - since this is just a numeric ID, might there be clashes with other preview URLs? Say that another app also has a record with `17` as ID?

Reply:
Yeah, that occurred to me too. I wondered whether it would still work with a content_id?

Reply:
To be honest, I don't really understand this part. If `content_id` would work, why would this have the complicated `z"%08x-%04x-%04x-%04x-%04x%08x" % ary` stuff at the bottom?

Could it be just `SecureRandom.uuid` instead?

Reply:
No Tijmen, you can't use `SecureRandom.uuid` because that'll change every time. This method is built to generate a _predictable_ string that is the same every time. So `content_id` will work as well.

Reply:
I'll change it to use the content_id.

Reply:
@tijmenb Updated and fixed-up